### DISCUSSION TITLE
File Search Stops Working

### DISCUSSION THREAD

**Discussion by ikeman32** [doublecmd/doublecmd#1522] [General]
I'm not sure if this is a problem with my system or with Double Commander. 
Double Commander
Version: 1.0.5 beta
Revision: 104
Commit: c02347051
Build date: 2022/04/12
Lazarus: 2.2.0.4
FPC: 3.2.2
Platform: x86_64-Linux-gtk2
OS version: Linux Mint 21.3 Virginia Cinnamon Desktop
Widgetset library: GTK 2.24.33

I have this very large set of text files that I will be using as a dataset for a GPT implementation I am working on as a personal project. The default file manager is unable to handle that many files, so I have been using Double Command to to search the folder for the text files. I stop the search when the count reaches 100K and click "Feed to listbox" then move the files 10K at a time to their own folders. Then I rinse and repeat, the problem is that periodically DC simply stops searching for the files and I have to wait an indeterminate amount of time before I can continue the process. But this only appears to happen on that one folder with all the text files. Nothing freezes up and DC continues to work fine otherwise I can even search other folders.

I have 16GB of RAM and a 4GB Swap Partition and they appear to not be affected. Memory and CPU usage is negligible consuming on average 39% of a single CPU core and 4.6GB of RAM. So the system doesn't appear to be stressed. 100K files in the listbox does seem to be close to the upper limit that it can handle as selecting the files is sluggish until the number of files is around 70K or fewer.

Also a reboot does not remedy the problem.

**Comment by JonnyTech** [comment]
```
Version: 1.0.5 beta
Build date: 2022/04/12
```

That is a very old version. Please try with the latest v1.1 builds.

**Comment by ikeman32** [comment]
I updated to the latest version as you suggested and the problem persists. However I did manage to find the xsession-errors in my home folder and found this error:
Window manager warning: Buggy client sent a _NET_ACTIVE_WINDOW message with a timestamp of 0 for 0x4600046

After a search for this error it seems to be pretty common  issue across multiple applications. This is apparently an old bug in the XIM  module of of Xlib: https://bugs.freedesktop.org/show_bug.cgi?id=39367

There was a work around a dev found with Eclipse: https://git.eclipse.org/c/platform/eclipse.platform.swt.git/commit/?id=bb4b85528d973e7f97e2a0b0f632b4f0f8401b9a

And my distro apparently uses a deprecated version of GDK, but when I switch to the QT version of DC the error messages still occur in my xsession-error; however, QT doesn't seem to care as I am able to continue with my project for now. If the problem returns I will switch to the appimage version and see if that corrects the problem or is only temporary.

If there are any other logs you think I should check or want to see I can provide them.  I will update if there are any other developments.

**Comment by ikeman32** [comment]
The QT version worked without a hitch and is apparently not affected by the bug in Xlib. But the gtk version is apparently.
